src/main/java/com/alex/eshop/eshop/Repository/ItemRepository.java

```diff
@@ -0,0 +1,22 @@
+package com.alex.eshop.eshop.Repository;
+
+import com.alex.eshop.eshop.Entity.Item;
+
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface ItemRepository extends JpaRepository<Item, Integer> {
+    @Query("select i from Item i order by i.id desc")
+    List<Item> getLastFiveItems();
```

Comment:
Good.
One problem: `List<Item> getLastFiveItems();` will return all the rows from table `item`. So this will be time and resource consuming. This is very bad.

Better instead of `itemRepository.getLastFiveItems().subList(0,5);` add new input parameter `Pageable pageable` to `getLastFiveItems();` then use `itemRepository.getLastFiveItems(PageRequest.of(0, 5));`. In this case you will get only 5 rows from database, not all.

You can also remove `List<Item> getLastFiveItems();` and use `Page<T> findAll(Pageable pageable);` from `interface PagingAndSortingRepository`.

result will be next:
```
public List<Item> getLastFiveItems() {
    return itemRepository.findAll(PageRequest.of(0, 5, Sort.Direction.DESC, "id")).getContent();
}
```